easypermissions/src/main/java/pub/devrel/easypermissions/EasyPermissions.java

```diff
@@ -247,8 +257,18 @@ private static void runAnnotatedMethods(Object object, int requestCode) {
 
     private static void checkCallingObjectSuitability(Object object) {
         // Make sure Object is an Activity or Fragment
-        if (!((object instanceof Fragment) || (object instanceof Activity))) {
-            throw new IllegalArgumentException("Caller must be an Activity or a Fragment.");
+        boolean isActivity = object instanceof Activity;
+        boolean isSupportFragment = object instanceof Fragment;
+        boolean isAppFragment = object instanceof android.app.Fragment;
+
+        if (!(isSupportFragment || isActivity ||
+                (isAppFragment && android.os.Build.VERSION.SDK_INT >= 23))) {
+            if (isAppFragment) {
+                throw new IllegalArgumentException(
```

Comment:
Thanks for the review. Please help me understand - 

The `hasPermissions` function accepts a `Context` and directly calls `ContextCompat.checkSelfPermission` with the `Context`. Shouldn't the `ContextCompat` have handled it already? I checked for an application with `targetSdkVersion` `22` and it returned `true` correctly. (This can be an optimization though).

As for the `shouldShowRequestPermissionRationale` method, it won't be invoked even if someone tries to call `requestPermissions`, as the the first line in `requestPermissions` invokes `checkCallingObjectSuitability`, where it throws `Exception` with a suitable message.